drivers/gpio/gpio_gecko.c

```diff
@@ -366,7 +366,7 @@ DEVICE_DT_DEFINE(DT_INST(0, silabs_gecko_gpio),
 		    gpio_gecko_common_init,
 		    NULL,
 		    &gpio_gecko_common_data, &gpio_gecko_common_config,
-		    POST_KERNEL, CONFIG_GPIO_GECKO_COMMON_INIT_PRIORITY,
+		    EARLY, CONFIG_GPIO_GECKO_COMMON_INIT_PRIORITY,
```

Comment:
EARLY level is meant for code that needs to run at very early stages, just after C domain is ready. Adjust priorities so that GPIO comes before UART instead. Or, if UART driver needs GPIO clocks, make sure it turns them on (better probably since you have an implicit dependency that is not captured anywhere).

Reply:
Changed the initialization stage from `EARLY` to `PRE_KERNEL_1` (the same as UART driver). Default INIT_PRIORITY for common part of the GPIO driver has priority `39` while for the UART driver it is `50` (priorities remain unchanged).